kauri/tree.go

```diff
@@ -0,0 +1,170 @@
+package kauri
+
+import (
+	"math"
+
+	"github.com/relab/hotstuff"
+)
+
+// MaxChild determines the branching factor of the tree.
+const MaxChild = 2
+
+// TreeConfiguration is an abstraction for a tree communication model.
+type TreeConfiguration interface {
+	InitializeWithPIDs(ids map[hotstuff.ID]int)
+	GetHeight() int
+	GetChildren() []hotstuff.ID
+	GetSubTreeNodes() []hotstuff.ID
+	GetParent() (hotstuff.ID, bool)
+}
+
+// FaultFreeTree implements a fault free tree configuration.
+type FaultFreeTree struct {
+	ID                  hotstuff.ID
+	ConfigurationLength int
+	height              int
+	idToPosMapping      map[hotstuff.ID]int
+	posToIDMapping      map[int]hotstuff.ID
+}
+
+// CreateTree Creates the tree configuration, currently only fault free tree configuration is supported.
```

Comment:
Renaming this function to `newFaultFreeTree` this doc comment can be replaced with:
*newFaultFreeTree creates a fault free tree configuration.*

There is no need to say that it is the only supported tree type.